api/v1/aerospikecluster_validating_webhook.go

```diff
@@ -263,6 +273,37 @@ func (c *AerospikeCluster) validate(aslog logr.Logger) error {
 	return c.validateSCNamespaces()
 }
 
+func (c *AerospikeCluster) validateOperation() error {
+	// Nothing to validate if no operation
+	if len(c.Spec.Operations) == 0 {
+		return nil
+	}
+
+	if c.Status.AerospikeConfig == nil {
+		return fmt.Errorf("operation cannot be set on create")
+	}
+
+	if len(c.Spec.Operations) > 1 {
+		return fmt.Errorf("only one operation can be set at a time")
+	}
+
+	allPodNames := GetAllPodNames(c.Name, c.Spec.Size, c.Spec.RackConfig.Racks)
+
+	podSet := sets.NewString(c.Spec.Operations[0].PodList...)
+	if !allPodNames.IsSuperset(sets.Set[string](podSet)) {
+		return fmt.Errorf("invalid pod name in operation")
+	}
+
+	// Don't allow any operation along with cluster scale up or racks added or removed
+	// New pods won't be available for operation
+	if !reflect.DeepEqual(c.Spec.Operations, c.Status.Operations) &&
```

Comment:
To prevent scaling up and adding operations together, a strict check is implemented. The operation ID is available in the status as soon as an on-demand operation is executed on any pod. However, this does not indicate that the operation is completed or that scaling up is allowed.